### PR TITLE
[OpenTelemetry.Extensions] Upgrade OpenTelemetry to latest1.4.0-beta.2

### PR DESCRIPTION
**Pull request by jvmlet**
## Changes

Latest `OpenTelemetry `API brings `System.Diagnostics.DiagnosticSource` version `7.0.0`.
The `ActivityEvent::Tags` property was redefined to be of type `Activity.TagsLinkedList(tags)` rather than `ActivityTagsCollection`, which makes impossible to add key/value to the dictionary once the the `ActivityEvent` was created.
This PR initializes the `ActivityEvent` **after** all tags had been added to `ActivityTagsCollection`

### CODE REVIEW

File: src/OpenTelemetry.Extensions/OpenTelemetry.Extensions.csproj [hunk 1]
```diff
@@ -16,7 +16,7 @@
   </ItemGroup>
 
   <ItemGroup>
-    <PackageReference Include="OpenTelemetry" Version="1.1.0" />
+    <PackageReference Include="OpenTelemetry" Version="1.4.0-beta.3.8" />
```
Comment (Kielek) [inline src/OpenTelemetry.Extensions/OpenTelemetry.Extensions.csproj:19]: There is not OpenTelemetry 1.4.0-beta.3.8 on the nuget side.

The latest stable is 1.3.1, the latest beta is 1.4.0-beta.1.

See: https://www.nuget.org/packages/OpenTelemetry/

File: test/OpenTelemetry.Extensions.Tests/ActivityEventAttachingLogProcessorTests.cs [hunk 2]
```diff
@@ -32,9 +32,9 @@ public sealed class ActivityEventAttachingLogProcessorTests : IDisposable
         ShouldListenTo = source => true,
     };
 
-    private readonly ILogger logger;
-    private readonly ILoggerFactory loggerFactory;
-    private OpenTelemetryLoggerOptions options;
+
```
Comment (Kielek) [inline test/OpenTelemetry.Extensions.Tests/ActivityEventAttachingLogProcessorTests.cs:35]: Could you please just remove empty lines?

**Comment by jvmlet** [comment]
@Kielek , please have a look at [NuGet.config](https://github.com/open-telemetry/opentelemetry-dotnet-contrib/blob/main/NuGet.config), it uses `MyGet` source, the latest version is `1.4.0-beta.3.8` and not `1.4.0-beta.1` from `NuGet` source. 
Tests pass with  `1.4.0-beta.3.8`

**Comment by Kielek** [comment]
@jvmlet, II suppose that this package should be published to nuget.org.
If so, the version used as a reference should also be available in this repository.

@open-telemetry/dotnet-contrib-maintainers, am I right?

**Comment by cijothomas** [comment]
> @jvmlet, II suppose that this package should be published to nuget.org. If so, the version used as a reference should also be available in this repository.
> 
> @open-telemetry/dotnet-contrib-maintainers, am I right?

we'll need to rely on the versions available in nuget.org.

**Comment by jvmlet** [comment]
@cijothomas , then I suppose you need to remove `  <add key="MyGet" value="https://www.myget.org/F/opentelemetry/api/v3/index.json" />` source declaration from `Nuget.config` file.

**Comment by jvmlet** [comment]
Can someone please explain how [1.4.0-beta.3.8](https://www.myget.org/feed/opentelemetry/package/nuget/OpenTelemetry/1.4.0-beta.3.8) was published **September 9 2202** and  [1.4.0-beta.2](https://www.myget.org/feed/opentelemetry/package/nuget/OpenTelemetry/1.4.0-beta.2) was published **October 17 2022** ?
See package history [here](https://www.myget.org/feed/opentelemetry/package/nuget/OpenTelemetry)

**Comment by cijothomas** [comment]
> @cijothomas , then I suppose you need to remove ` <add key="MyGet" value="https://www.myget.org/F/opentelemetry/api/v3/index.json" />` source declaration from `Nuget.config` file.

Do we have to remove it? There are situations where some component want to use a new feature only available in the nightly myget build. Agree we haven't seen such cases recently, so probably a good idea to cleanup config and have it only point to nuget.org

**Comment by jvmlet** [comment]
@cijothomas
These 2 comments are mutually exclusive :
>we'll need to rely on the versions available in nuget.org.

>There are situations where some component want to use a new feature only available in the nightly myget build.

I still can't understand why version`build.3.8` was published before `build.2`

**Comment by cijothomas** [comment]
> Can someone please explain how [1.4.0-beta.3.8](https://www.myget.org/feed/opentelemetry/package/nuget/OpenTelemetry/1.4.0-beta.3.8) was published **September 9 2202** and [1.4.0-beta.2](https://www.myget.org/feed/opentelemetry/package/nuget/OpenTelemetry/1.4.0-beta.2) was published **October 17 2022** ? See package history [here](https://www.myget.org/feed/opentelemetry/package/nuget/OpenTelemetry)

Don't know. Its irrelevant to this PR anyway, and can be separately checked.

**Comment by jvmlet** [comment]
> Don't know. Its irrelevant to this PR anyway, and can be separately checked.

Well, this  **is** relevant, because it relys on API available in `3.8`, anyway, downgraded to the latest `beta.2` from `nuget.org`

**Review by cijothomas — APPROVED** [review]

**Comment by cijothomas** [comment]
@CodeBlanch please review when you get a chance!

**Review by Kielek — APPROVED** [review]

**Comment by jvmlet** [comment]
Can someone please have a look at flaky tests?

**Review by CodeBlanch — APPROVED** [review]
LGTM! @jvmlet would you mind updating CHANGELOG with a note + link to PR?

**Comment by jvmlet** [comment]
@dotnet-contrib-maintainers, please release

**Comment by jvmlet** [comment]
still flaky

File: src/OpenTelemetry.Extensions/CHANGELOG.md [hunk 3]
```diff
@@ -12,6 +12,10 @@
   can still be used with .NET5.0 apps.
   ([#617](https://github.com/open-telemetry/opentelemetry-dotnet/pull/617))
 
+## 1.0.0-beta.4
+
+* Support latest  OpenTelemetry to latest1.4.0-beta.2 ([#680](https://github.com/open-telemetry/opentelemetry-dotnet-contrib/pull/680))
```
Comment (cijothomas) [inline src/OpenTelemetry.Extensions/CHANGELOG.md:17]: the wording looks odd.

**Comment by jvmlet** [comment]
can we please merge ?

File: src/OpenTelemetry.Extensions/CHANGELOG.md [hunk 4]
```diff
@@ -12,6 +12,10 @@
   can still be used with .NET5.0 apps.
   ([#617](https://github.com/open-telemetry/opentelemetry-dotnet/pull/617))
 
+## 1.0.0-beta.4
```
Comment (Kielek) [inline src/OpenTelemetry.Extensions/CHANGELOG.md:15]: I think that you should take care about changelog once more time. It should be part of unrelesed or all from unreleaset should be here.

File: src/OpenTelemetry.Extensions/OpenTelemetry.Extensions.csproj [hunk 5]
```diff
@@ -16,7 +16,7 @@
   </ItemGroup>
 
   <ItemGroup>
-    <PackageReference Include="OpenTelemetry" Version="1.1.0" />
+    <PackageReference Include="OpenTelemetry" Version="1.4.0-beta.2" />
```
Comment (Kielek) [inline src/OpenTelemetry.Extensions/OpenTelemetry.Extensions.csproj:19]: Please use variables from: https://github.com/open-telemetry/opentelemetry-dotnet-contrib/pull/752

Comment (jvmlet) [reply src/OpenTelemetry.Extensions/OpenTelemetry.Extensions.csproj]: fixed

File: src/OpenTelemetry.Extensions/CHANGELOG.md (hunk 4 above)
Comment (jvmlet) [reply src/OpenTelemetry.Extensions/CHANGELOG.md]: fixed

**Comment by jvmlet** [comment]
Still flaky

File: src/OpenTelemetry.Extensions/CHANGELOG.md (hunk 4 above)
Comment (utpilla) [reply src/OpenTelemetry.Extensions/CHANGELOG.md]: @jvmlet You have updated the CHANGELOG to add an entry for `1.4.0-beta.4`. Did you want to have the `1.4.0-beta.4` version of this package released?

Comment (jvmlet) [reply src/OpenTelemetry.Extensions/CHANGELOG.md]: Yes  please

**Comment by utpilla** [comment]
I have published the package: https://www.nuget.org/packages/OpenTelemetry.Extensions/1.0.0-beta.3

**Pull request merged**